api/ipam.go

```diff
@@ -5,6 +5,11 @@ import (
 	"net"
 )
 
+// Special "I don't have a unique ID" container identifier, used in various cases:
+// - when the Docker plugin API does not make the container ID available at the point Weave needs it,
+// - when reclaiming an IP for a container using a different network interface.
```

Comment:
Rather than trying to enumerate the cases, which will be hard to keep up to date, make a more general statement "when the caller does not know the container ID, or is unsure whether IPAM was previously told the ID".

I think the point about the address being filed under its own string, so that needs to be used when deleting, should also be brought out here.